app/controllers/hosts_controller.rb

```diff
@@ -791,6 +792,15 @@ def load_vars_for_ajax
     end
   end
 
+  def validate_host_owner
```

Comment:
In reality this validation should go in the Hostext::Ownership concern, so that it's applied every time in the model. Otherwise it's just brittle and it'll only validate under whatever controller scenario. Please move it there - the controller already knows how to respond to a failed validation & display the error.